pkg/controller/controller.go

```diff
@@ -1060,8 +1101,9 @@ func convertCatalog(in *brokerapi.Catalog) ([]*v1alpha1.ServiceClass, error) {
 		if svc.Metadata != nil {
 			metadata, err := json.Marshal(svc.Metadata)
 			if err != nil {
-				glog.Errorf("Failed to unmarshal metadata\n%+v\n %v", svc.Metadata, err)
-				return nil, fmt.Errorf("Failed to unmarshal metadata\n%+v\n %v", svc.Metadata, err)
+				err = fmt.Errorf("Failed to marshal metadata\n%+v\n %v", svc.Metadata, err)
```

Comment:
for other reviewers, we don't need to say which broker we're on because this error will be wrapped by another one that'll have that info.